frontend/src/components/MapView/LeftPanel/layersPanel/MenuSwitch/SwitchItem/LayerDownloadOptions.tsx

```diff
@@ -35,16 +35,21 @@ import { useSafeTranslation } from 'i18n';
 import { isExposureAnalysisLoadingSelector } from 'context/analysisResultStateSlice';
 import { availableDatesSelector } from 'context/serverStateSlice';
 import { getRequestDate } from 'utils/server-utils';
+import { LayerDefinitions } from 'config/utils';
 
 // TODO - return early when the layer is not selected.
 function LayerDownloadOptions({
-  layer,
+  layerId,
   extent,
   selected,
   size,
 }: LayerDownloadOptionsProps) {
   const { t } = useSafeTranslation();
   const dispatch = useDispatch();
+  const layer = useMemo(() => {
```

Comment:
`useMemo` may actually be slowing us down here, since the calculation involved is not that expensive. #901